src/shogun/distance/Distance.h

```diff
@@ -335,6 +335,28 @@ class CDistance : public CSGObject
 			return lhs==rhs;
 		}
 
+		/**
+		 * Function for computing distance values
+		 *
+		 * @param result array of distance values
+		 * @param idx_r_start 
+		 * @param idx_start
+		 * @param idx_stop
+		 * @param idx_a feature vector a at idx_a
+		 */
+		float64_t* run_distance_rhs(float64_t* result, int32_t idx_r_start, int32_t idx_start, int32_t idx_stop, int32_t idx_a);
```

Comment:
why do you pass native `float64_t*` arrays, when we have `SGVector<float64_t>` for such things...